impeller/toolkit/egl/display.cc

```diff
@@ -136,6 +136,10 @@ std::unique_ptr<Config> Display::ChooseConfig(ConfigDescriptor config) const {
     attributes.push_back(static_cast<EGLint>(config.stencil_bits));
   }
 
+  // On android emulators multisample buffer configs don't work, so we fallback
```

Comment:
The stuff in //impeller/toolkit is meant to be platform agnostic. If a `Samples::kFour` config isn't found, this method will return `nullptr` to the caller. Can the caller can then try a different config? That way, this Android quirk doesn't affect other platforms.

I patched the Android stuff like so (don't know if it compiles but just a gist):

```
diff --git a/shell/platform/android/android_surface_gl_impeller.cc b/shell/platform/android/android_surface_gl_impeller.cc
index 3200822776..86437bb945 100644
--- a/shell/platform/android/android_surface_gl_impeller.cc
+++ b/shell/platform/android/android_surface_gl_impeller.cc
@@ -100,8 +100,17 @@ AndroidSurfaceGLImpeller::AndroidSurfaceGLImpeller(
   desc.surface_type = impeller::egl::SurfaceType::kWindow;
   auto onscreen_config = display->ChooseConfig(desc);
   if (!onscreen_config) {
-    FML_DLOG(ERROR) << "Could not choose onscreen config.";
-    return;
+    // This is probably an Android simulator. Fallback to a single sample so we
+    // get something up on the screen.
+    desc.samples = impeller::egl::Samples::kOne;
+    onscreen_config = display->ChooseConfig(desc);
+    if (onscreen_config) {
+      FML_LOG(INFO) << "Warning: This device doesn't support MSAA for onscreen "
+                       "framebuffers. Falling back to a single sample.";
+    } else {
+      FML_DLOG(ERROR) << "Could not choose onscreen config.";
+      return;
+    }
   }
 
   desc.surface_type = impeller::egl::SurfaceType::kPBuffer;

```

We actually don't want to use kOne because that would disable MSAA on the onscreen FBO. We are just doing it so something shows up on simulators. Let's do that only on Android and log a message when we do so we know something is expected to not work. Instead of silently breaking like here.